core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java

```diff
@@ -172,16 +222,60 @@ String getExperimentFeatureKey(String experimentId) {
 
         Map<String, OptimizelyFeature> optimizelyFeatureKeyMap = new HashMap<>();
         for (FeatureFlag featureFlag : featureFlags) {
-            optimizelyFeatureKeyMap.put(featureFlag.getKey(), new OptimizelyFeature(
+            Map<String, OptimizelyExperiment> experimentsMapForFeature =
+                getExperimentsMapForFeature(featureFlag.getExperimentIds(), allExperimentsMap);
+            OptimizelyFeature optimizelyFeature = new OptimizelyFeature(
                 featureFlag.getId(),
                 featureFlag.getKey(),
                 getExperimentsMapForFeature(featureFlag.getExperimentIds(), allExperimentsMap),
                 getFeatureVariablesMap(featureFlag.getVariables())
-            ));
+            );
+
+            List<OptimizelyExperiment> experimentRules =
+                new ArrayList<OptimizelyExperiment>(experimentsMapForFeature.values());
+            List<OptimizelyExperiment> deliveryRules =
+                this.getDeliveryRules(this.projectConfig.getRollouts(), featureFlag.getRolloutId());
+
+            optimizelyFeature.setDeliveryRules(deliveryRules);
+            optimizelyFeature.setExperimentRules(experimentRules);
```

Comment:
Same here. Can we move these to constructor parameters?

Reply:
Since Java does not support default parameters - Perhaps at a later date we can refactor to use the builder design pattern. May help for maintainability down the road when adding new parameters so we do not end up with huge constructors like we are already seeing. Let me know what you think. This would be a good solution in my opinion.

Reply:
It may be a good idea. It's expected we will need to extend the API again later.